diff --git a/node_modules/@google/genai/dist/genai.d.ts b/node_modules/@google/genai/dist/genai.d.ts
index 7c7be1a..d6de3ee 100644
--- a/node_modules/@google/genai/dist/genai.d.ts
+++ b/node_modules/@google/genai/dist/genai.d.ts
@@ -3416,6 +3416,10 @@ export declare interface GoogleGenAIOptions {
      * Optional. A set of customizable configuration for HTTP requests.
      */
     httpOptions?: HttpOptions;
+    /**
+     * Optional. Custom fetch implementation.
+     */
+    fetch?: typeof fetch;
 }
 
 /** Tool to support Google Maps in Model. */
diff --git a/node_modules/@google/genai/dist/node/index.cjs b/node_modules/@google/genai/dist/node/index.cjs
index 3e72754..0d8af94 100644
--- a/node_modules/@google/genai/dist/node/index.cjs
+++ b/node_modules/@google/genai/dist/node/index.cjs
@@ -10626,6 +10626,11 @@ class ApiClient {
         if (opts.httpOptions) {
             this.clientOptions.httpOptions = this.patchHttpOptions(initHttpOptions, opts.httpOptions);
         }
+        this.fetch = fetch;
+        if (opts.fetch) {
+            console.log('@google/genai is using custom fetch implementation.');
+            this.fetch = opts.fetch;
+        }
     }
     /**
      * Determines the base URL for Vertex AI based on project and location.
@@ -10946,7 +10951,7 @@ class ApiClient {
         });
     }
     async apiCall(url, requestInit) {
-        return fetch(url, requestInit).catch((e) => {
+        return this.fetch(url, requestInit).catch((e) => {
             throw new Error(`exception ${e} sending request`);
         });
     }
@@ -15819,6 +15824,7 @@ class GoogleGenAI {
             userAgentExtra: LANGUAGE_LABEL_PREFIX + process.version,
             uploader: new NodeUploader(),
             downloader: new NodeDownloader(),
+            fetch: options.fetch,
         });
         this.models = new Models(this.apiClient);
         this.live = new Live(this.apiClient, auth, new NodeWebSocketFactory());
diff --git a/node_modules/@google/genai/dist/node/index.mjs b/node_modules/@google/genai/dist/node/index.mjs
index 4a0ef41..d74333d 100644
--- a/node_modules/@google/genai/dist/node/index.mjs
+++ b/node_modules/@google/genai/dist/node/index.mjs
@@ -10604,6 +10604,11 @@ class ApiClient {
         if (opts.httpOptions) {
             this.clientOptions.httpOptions = this.patchHttpOptions(initHttpOptions, opts.httpOptions);
         }
+        this.fetch = fetch;
+        if (opts.fetch) {
+            console.log('@google/genai is using custom fetch implementation.');
+            this.fetch = opts.fetch;
+        }
     }
     /**
      * Determines the base URL for Vertex AI based on project and location.
@@ -10924,7 +10929,7 @@ class ApiClient {
         });
     }
     async apiCall(url, requestInit) {
-        return fetch(url, requestInit).catch((e) => {
+        return this.fetch(url, requestInit).catch((e) => {
             throw new Error(`exception ${e} sending request`);
         });
     }
@@ -15797,6 +15802,7 @@ class GoogleGenAI {
             userAgentExtra: LANGUAGE_LABEL_PREFIX + process.version,
             uploader: new NodeUploader(),
             downloader: new NodeDownloader(),
+            fetch: options.fetch,
         });
         this.models = new Models(this.apiClient);
         this.live = new Live(this.apiClient, auth, new NodeWebSocketFactory());
